core/dbt/clients/registry.py

```diff
@@ -34,9 +34,33 @@ def _get_url(name, registry_base_url=None):
     return "{}{}".format(registry_base_url, url)
 
 
-def _get_with_retries(package_name, registry_base_url=None):
+def _check_package_redirect(package_name: str, response: Dict[str, Any]) -> None:
+    # Either redirectnamespace or redirectname in the JSON response indicate a redirect
+    # redirectnamespace redirects based on package ownership
+    # redirectname redirects based on package name
+    # Both can be present at the same time, or neither. Fails gracefully to old name
+    if ("redirectnamespace" in response) or ("redirectname" in response):
+        if ("redirectnamespace" in response) and response["redirectnamespace"] is not None:
+            use_namespace = response["redirectnamespace"]
+        else:
+            use_namespace = response["namespace"]
+
+        if ("redirectname" in response) and response["redirectname"] is not None:
+            use_name = response["redirectname"]
+        else:
+            use_name = response["name"]
+
+        new_nwo = use_namespace + "/" + use_name
```

Comment:
```suggestion
        new_nwo = f"{use_namespace}/{use_name}"
```

Reply:
Another good thing to get updated, nice catch! ❤️